TheMET/TheMET/Services/FeaturedArtService.swift

```diff
@@ -127,34 +125,43 @@ class FeaturedArtService {
         self.featuredArt = .loading
         self.metAPI.objects { [weak self] objectResponce in
             guard let objectResponce = objectResponce,
-            let randomId = objectResponce.objectIDs.randomElement() else {
+                  let randomId = self?.getRandomID(objectToDelete: nil, objectResponce: objectResponce) else {
                 self?.featuredArt = .failed
                 self?.isFeaturedArtLoading = false
                 return
             }
-            self?.metAPI.object(id: randomId) {[weak self] object in
-                guard let object = object else {
+            self?.tryRandomId(randomID: randomId, objectResponce: objectResponce)
+        }
+    }
+    
+    private func getRandomID (objectToDelete: ArtID?, objectResponce: ObjectsResponse) -> ArtID? {
+        objectResponce.objectIDs.removeAll { $0 == objectToDelete }
```

Comment:
You remove only a single object-id even if there were multiple attempts.
try next:

- implement method `updateFeaturedArt(objectIDs: [ArtID])`. It supposed to be executed by `forceUpdateFeaturedArt` when the `objectsResponce` is received.

- download a random object using `self.metAPI.object(id:completion:)` inside `updateFeaturedArt(objectIDs:)`. If it is invalid, create a new array by modifying `objectIDs`-array and execute `updateFeaturedArt(objectIDs:)` again with a new argument.